### DISCUSSION TITLE
Sending multiple documents types from API Loader into the same vector store

### DISCUSSION THREAD

**Discussion by poniatowski1987** [FlowiseAI/Flowise#1671] [Q&A] [answered]
Hi, I would like to be able to load multiple documents via separate API Loader nodes (e.g. a PDF document with text content, and a separate JSON with version history of the same document) and query them using Conversational Retrieval QA Chain. Importantly, I want to use an in-memory vector storage.

To achieve this, I attempt to use a flow like the one below. I can see that in GUI, the In-Memory Vector Store lets me connect multiple API Loader nodes to it:
<img width="1188" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/68443026/48d20e80-4e6c-44ce-8f59-b25c62a406d5">

However, any prompt I try to send is only able to use the data from the first (top, PDF) document sent from the first API Loader node. But I know for sure that the second node (bottom, JSON) is not the issue - because when I disconnect the first node from the flow and only leave the second one, it understand the JSON and I am able to run prompts against it successfully, like this:
<img width="1336" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/68443026/49d57759-7ec8-451a-a8f6-e82de97e1605">

Across the tutorials I found, I understand one can load multiple PDFs via multiple PDF File loader nodes, then add metadata to each node separately before sending to a vector db - but that is not an option I see in an API Loader node.

Can you please guide me on what am I missing? Perhaps the flow should be built differently to reach the same result?

Many thanks in advance

**Comment by HenryHengZJ** [comment, marked as answer]
what I usually recommend others doing is to use a vector store with metadata. Upsert specific data with different metadata key value pair.
Later at query time, use Retriever Tool with description to let LLM knows which vector store with specific metadata to query:
![image](https://github.com/FlowiseAI/Flowise/assets/26460777/8a0dedb8-35a6-46fd-a17d-f9e62cac897c)

**Reply by poniatowski1987** [reply]
Thanks a ton for your answer, @HenryHengZJ! I played around a bit more. Is there a way to inject metadata when I'm loading documents through an APILoader node, and upserting to a database like Pinecone? In a "Pdf file" node I can add them under "Additional Parameters", but I don't think I can do that through APILoader, unless there is some other way. 
<img width="734" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/68443026/43a2a143-a731-41f6-9d83-3e417d7e39af">

**Reply by chungyau97** [reply]
Hey @poniatowski1987 
This is the config you need to send in your body
```
{
    "question": "What's Jane?",
    "overrideConfig": {
        "supabaseMetadataFilter": {
            "supabaseExistingIndex_0": {
                "user": "jane",
                "docname": "janedoc"
            }
        }
    }
}
```
You can read more [here](https://docs.flowiseai.com/using-flowise/api).
